qoajad_backend/src/main/java/com/qoajad/backend/controller/AppointmentController.java

```diff
@@ -26,38 +25,53 @@ public AppointmentController(@Qualifier("defaultAppointmentService") final Appoi
-        } catch (EmptyResultDataAccessException e) {
-            response = new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
+    @RequestMapping(value = "/appointment/all_available_appointments/{hpiName}/{specialtyName}", method = RequestMethod.GET)
+    public ResponseEntity<List<ConsultingRoom>> findAvailableAppointments(@PathVariable("hpiName") final String hpiName,
+                                                                          @PathVariable("specialtyName") final String specialtyName) {
+        ResponseEntity<List<ConsultingRoom>> response;
+        response = new ResponseEntity<>(appointmentService.findAvailableAppointments(hpiName, specialtyName), HttpStatus.OK);
+        return response;
+    }
+
+    @RequestMapping(value = "/appointment/all_user_appointments/{userDocument}", method = RequestMethod.GET)
+    public ResponseEntity<List<Appointment>> findUserAppointments(@PathVariable("userDocument") final int userDocument) {
+        ResponseEntity<List<Appointment>> response;
+        response = new ResponseEntity<>(appointmentService.findUserAppointments(userDocument), HttpStatus.OK);
+
         return response;
     }
 
     @RequestMapping(value = "/appointment/create", method = RequestMethod.POST)
-    public ResponseEntity<String> createAppointment(@RequestBody CreateAppointment createAppointment, final HttpServletRequest request) {
+    public ResponseEntity<String> createAppointment(@RequestBody final CreateAppointment createAppointment) {
         ResponseEntity<String> response;
-        try {
-            appointmentService.createAppointment(createAppointment);
+        boolean rowsChanged;
+        rowsChanged = appointmentService.attemptToCreateAppointment(createAppointment);
+        if(rowsChanged) {
             response = new ResponseEntity<>("Appointment created successfully.", HttpStatus.CREATED);
-        } catch (DataIntegrityViolationException e) {
-            response = new ResponseEntity<>(HttpStatus.CONFLICT);
+        } else {
+            response = new ResponseEntity<>("An error has occurred while creating the appointment.", HttpStatus.CONFLICT);
         }
+
         return response;
     }
 
     @RequestMapping(value = "/appointment/update", method = RequestMethod.PUT)
-    public ResponseEntity<String> updateAppointment(@RequestBody UpdateAppointment updateAppointment, final HttpServletRequest request) {
+    public ResponseEntity<String> updateAppointment(@RequestBody final UpdateAppointment updateAppointment) {
         ResponseEntity<String> response;
-        try {
-            appointmentService.updateAppointment(updateAppointment);
-            response = new ResponseEntity<>("Appointment updated successfully.", HttpStatus.CREATED);
-        } catch (DataIntegrityViolationException e) {
-            response = new ResponseEntity<>(HttpStatus.CONFLICT);
-        }
+        int rowsChanged;
```

Comment:
```suggestion
```
Ditto. Dont split the int and ResponseEntity definition with the assignment if its not needed.